apps/client/src/hoc/withChecker.js

```diff
@@ -1,57 +1,55 @@
-import React, { useState, useEffect, useContext } from "react";
+import React, { useContext, useEffect, useState } from "react";
 
-import { SessionContext, CheckerContext } from "../context";
-import withAddress from "./withAddress";
-import getChecker from "../sttr_client";
-import LoadingPage from "../pages/LoadingPage";
+import { CheckerContext, SessionContext } from "../context";
 import ErrorPage from "../pages/ErrorPage";
+import LoadingPage from "../pages/LoadingPage";
+import getChecker from "../sttr_client";
+import withTopic from "./withTopic";
 
 const dir =
   process.env.REACT_APP_STTR_ENV === "production" ? "PROD" : "STAGING";
 
 const withChecker = (Component) =>
-  withAddress(({ ...rest }) => {
+  withTopic((props) => {
     const sessionContext = useContext(SessionContext);
     const checkerContext = useContext(CheckerContext);
     const [checker, setChecker] = useState(checkerContext.checker);
     const [error, setError] = useState();
-
-    const { topic } = rest;
-
-    useEffect(() => {
-      // Load new `sttr-checker` file and store in the Context
-      if (!checker && !error) {
-        fetch(
-          `${window.location.origin}/sttr/${dir.toLowerCase()}/${
-            topic.sttrFile
-          }`
-        )
-          .then((response) => response.json())
-          .then((json) => {
-            const checker = getChecker(json);
-            if (sessionContext.answers) {
-              checker.setQuestionAnswers(sessionContext.answers);
-              // In case of reload, rewind to the current question
-              checker.rewindTo(sessionContext.questionIndex);
-            } else {
-              checker.next();
-            }
-
-            // Store the entire `sttr-checker` in React Context
-            checkerContext.checker = checker;
-            setChecker(checker);
-          })
-          .catch((e) => {
-            setError(e);
-          });
-      }
-    });
+    const { topic } = props;
+    const { sttrFile } = topic;
+
+    if (sttrFile) {
+      useEffect(() => {
+        if (!checker && !error) {
+          fetch(
+            `${window.location.origin}/sttr/${dir.toLowerCase()}/${sttrFile}`
+          )
+            .then((response) => response.json())
+            .then((json) => {
+              const newChecker = getChecker(json);
+              if (sessionContext.answers) {
+                newChecker.setQuestionAnswers(sessionContext.answers);
```

Comment:
Should we not change the way we store things in session, like sessionContext[slug].answers? Because now it looks like answers are only stored once. Better in a next PR?

Reply:
Yes! We should definitely fix this! Thank you.